runtime/ast/expression.go

```diff
@@ -311,6 +311,19 @@ func NewFixedPointExpression(
 	scale uint,
 	tokenRange Range,
 ) *FixedPointExpression {
+	common.UseMemory(
+		gauge,
+		common.NewBigIntMemoryUsage(
+			common.BigIntByteLength(integer),
+		),
+	)
+	common.UseMemory(
+		gauge,
+		common.NewBigIntMemoryUsage(
+			common.BigIntByteLength(fractional),
+		),
+	)
```

Comment:
This metering of the `integer` and `fractional` arguments passed to this function should be moved to the call-site, before those arguments are allocated.

Was the concern here that those come from the lexer/tokens?

Reply:
Yes, the concern was that we don't know the size until the creation. And the assumption was that no single bigInt can leapfrog the memory safety-band

Reply:
Don't we have to meter the big-int construction in `parseFixedPointPart`? I assumed we would need something like an inverse of `OverEstimateNumberStringLength`

Reply:
Updated in 4b717cf